src/main/java/com/redislabs/redisgraph/RedisGraphAPI.java

```diff
@@ -96,13 +106,178 @@ public String deleteGraph() {
 		}
     }
 
+
+    /**
+     * Sends command
+     * @param conn
+     * @param provider
+     * @param args
+     * @return
+     */
     private BinaryClient sendCommand(Jedis conn, ProtocolCommand provider, String ...args) {
         BinaryClient binaryClient = conn.getClient();
         binaryClient.sendCommand(provider, args);
         return binaryClient;
     }
+
+
+    /**
+     * Sends the command with --COMPACT flag
+     * @param conn
+     * @param provider
+     * @param args
+     * @return
+     */
+    private BinaryClient sendCompactCommand(Jedis conn, ProtocolCommand provider, String ...args) {
+        BinaryClient binaryClient = conn.getClient();
+        List<String> largs = new ArrayList<>(Arrays.asList(args));
+        largs.add("--COMPACT");
+        String[] t = new String[largs.size()];
+        binaryClient.sendCommand(provider, largs.toArray(t));
+        return binaryClient;
+    }
     
     private Jedis getConnection() {
         return this.client.getResource();
     }
+
+
+    /**
+     * Invokes stored procedures without arguments
+     * @param procedure procedure name to invoke
+     * @return
+     */
+    public ResultSet callProcedure(String procedure  ){
+        return callProcedure(procedure, new ArrayList<>(), new HashMap<>());
```

Comment:
shouldn't " new ArrayList<>(), new HashMap<>()" be constant?

Reply:
If you just need a "NULL" List and Map I would have keep two Immutable static final dummy map and list and not create a new empty two on every call